doc/api-event-handling.md

```diff
@@ -0,0 +1,172 @@
+       defaultSubscriptionPolicy: ""
+       defaultInitialTerminationTime: "PT1H"
+       defaultAutoRenew: true
+       defaultTopicFilter: "tns1:RuleEngine/TamperDetector"
+       defaultMessageContentFilter: "boolean(//tt:SimpleItem[@Name=”IsTamper”])"
+       defaultMessageTimeout: "PT5S"
+       defaultMessageLimit: 10
+     properties:
+       valueType: "Object"
+       readWrite: "W"
+    ```
+
+* Define a UnsubscribeCameraEvent resource for unsubscribing
+    ```yaml
+    - name: "UnsubscribeCameraEvent"
+     isHidden: false
+     description: "Unsubscribe all event from the camera"
+     attributes:
+       service: "EdgeX"
+       setFunction: "UnsubscribeCameraEvent"
+     properties:
+       valueType: "Object"
+       readWrite: "W"
+    ```
+
+### Define device resource for BaseNotification
+
+* Define a SubscribeCameraEvent resource with BaseNotification subscribeType
+    ```yaml
+    - name: "SubscribeCameraEvent"
+     isHidden: false
+     description: "Create a subscription to subscribe the event ..."
+     attributes:
+       service: "EdgeX"
+       setFunction: "SubscribeCameraEvent"
+       # PullPoint | BaseNotification
+       subscribeType: "BaseNotification"
+       defaultSubscriptionPolicy: ""
+       defaultInitialTerminationTime: "PT1H"
+       defaultAutoRenew: true
+       defaultTopicFilter: "..."
+       defaultMessageContentFilter: "..."
+     properties:
+       valueType: "Object"
+       readWrite: "W"
+    ```
+
+* Define a driver config BaseNotificationURL to indicate the device service network location
+    ```
+    # configuration.toml
+    [Driver]
+    # BaseNotificationURL indicates the device service network location, the user must replace the host to match their machine
+    BaseNotificationURL = "http://192.168.12.112:59984"
+    ```
+
+Device service will generate the following path for pushing event from Camera to device service:
+- {BaseNotificationURL}/api/v2/resource/{DeviceName}/{ResourceName}
+- {BaseNotificationURL}/api/v2/resource/Camera1/CameraEvent
+
+**Note**: The user can also override the config from the docker-compose environment variable:
+```shell
+export HOST_IP=$(ifconfig eth0 | grep "inet " | awk '{ print $2 }')
+```
+```yaml
+environment:
+   DRIVER_BASENOTIFICATIONURL: http://${HOST_IP}:59984
+```
+Then the device service can be accessed by the external camera from the other subnetwork.
+
+### Define device resource for unsubscribing the event
+```yaml
+  - name: "UnsubscribeCameraEvent"
+    isHidden: true
+    description: "Unsubscribe all subscription from the camera"
+    attributes:
+      service: "EdgeX"
+      setFunction: "UnsubscribeCameraEvent"
+    properties:
+      valueType: "Object"
+      readWrite: "W"
+```
+
+## Find the supported Event Topics
+Finding out what notifications a camera supports and what information they contain:
+
+```shell
+curl --request GET 'http://0.0.0.0:59882/api/v2/device/name/Camera003/GetEventProperties'
```

Comment:
Note that the `ServerBindAddr` has to be set to `0.0.0.0` for this to work. Please add how to override SERVICE_SERVERBINDADDR for this and that it should only be use on local secure network as it opens the service up to access to from any thing on the network.

Reply:
Thanks for catching that, I consider using `localhost` instead of `0.0.0.0`, the `ServerBindAddr` feature should be introduced in another doc.